Some UI tweaks to the Dart problems view

devoncarew opened the pull request:
Some UI tweaks to the Dart problems view:

- in the filter dialog for the Dart problems view, rename `content root` to `module` (I think this maps closer to IntelliJ concepts the user would be familiar with)
- rename `package` to `Dart package`

<img width="378" alt="screen shot 2016-09-05 at 12 19 55 pm" src="https://cloud.githubusercontent.com/assets/1269969/18258200/06939292-7386-11e6-99b1-ca52a4e63bd2.png">

- if there are no issues in the problems view, hide the status label at the bottom
- change the summary text at the bottom to just show counts for the items displayed in the errors view, and if a filter is active, tell the user what we're filtering on

<img width="451" alt="screen shot 2016-09-05 at 12 18 50 pm" src="https://cloud.githubusercontent.com/assets/1269969/18258196/fd9a973a-7385-11e6-9c25-23abfb009a4c.png">

@alexander-doroshko @jwren

alexander-doroshko commented:
> rename `content root` to `module`

I'm afraid it is 
a) not true for IntelliJ IDEA (each module may have any number of content roots, including 0, 1 and >1)
b) not applicable for smaller IDEs because they do not have `module` concept at all.

[Content root](https://www.jetbrains.com/help/idea/content-root.html) is one of the basic concepts in any JetBrains IDE, and this term can be met in Project Structure dialog (IntelliJ IDEA) or in Preferences | Directories (other IDEs).

You should understand that for our IDEs there are no such things as 'project root' or 'module root'. There are:
- folder where IDE stores project files (.idea folder), and it is never of interest  for us
- IntelliJ IDEA project may contain zero or more modules. Smaller IDEs as a rule contain 1 module (that's why there's no `module` concept there)
- module settings are stored in *.iml file and location of this file is not connected to .idea folder, module roots, or whatever else, and never of interest for us
- each module may contain zero or more content roots, in general case location of these directories are absolutely independent and not related to anything else
- each content root may have (all or some, depending on the IDE) source folders, test source folders, resource folders, test resources, excluded folders. At any level inside content root.

So, 'project root' term makes any sense only when project contains one module and this module contains one content root (and usually this content root is the folder where .idea folder is stored). But we must not rely on any specific project structure and that's why we avoid 'project root' and 'module root' terms.

alexander-doroshko commented:
> rename `package` to `Dart package`

No objections if everyone agrees. May be do the same for filter status in the status text?

alexander-doroshko commented:
> if there are no issues in the problems view, hide the status label at the bottom

That's fine if there are no filters enabled. If there are issues, but all filtered out, I think it makes sense to say that filtering is enabled.

alexander-doroshko commented:
> change the summary text at the bottom

May be loop Luke as UX expert or our UX expert in JetBrains?
For the reference, here's what we compare:
![image](https://cloud.githubusercontent.com/assets/4499789/18270019/89b16e94-7433-11e6-990d-0f3d13a50f1d.png)

![image](https://cloud.githubusercontent.com/assets/4499789/18270088/c9d8b496-7433-11e6-9b94-fe2f7cdaf783.png)

devoncarew commented:
OK, I updated the PR to revert the content root => module filter text change. Sounds like if we wanted to filter at a module level, we'd need to accept files in any of the content roots for that module.

> That's fine if there are no filters enabled. If there are issues, but all filtered out, I think it makes sense to say that filtering is enabled.

Makes sense; I updated the PR to do that.

<img width="435" alt="screen shot 2016-09-06 at 11 40 39 am" src="https://cloud.githubusercontent.com/assets/1269969/18286527/1542b7bc-7428-11e6-8396-c3a0adbaf06c.png">

<img width="341" alt="screen shot 2016-09-06 at 11 40 56 am" src="https://cloud.githubusercontent.com/assets/1269969/18286528/1542d7e2-7428-11e6-9174-dce5976ddbd1.png">

> May be loop Luke as UX expert or our UX expert in JetBrains?

Happy to! @lukechurch for thoughts :)

I think the updated status line summary is a bit more terse, and a bit easier for users to parse. Most of what they want to know will be in the body of the problems view itself.

alexander-doroshko commented:
![image](https://cloud.githubusercontent.com/assets/4499789/18313401/518f1072-7517-11e6-9b18-77691288e8d9.png)
:)

alexander-doroshko commented:
[Pushed](https://github.com/JetBrains/intellij-plugins/commit/883fcbf1f121b18acff8ce02fd2ac86b651d95c7) manually (to avoid rename/revert in commits) + added fix for `( and severity)`.

devoncarew commented:
> added fix for ( and severity).

oops :)
